### PR TITLE
Separate fft normalizations from plans

### PR DESCRIPTION
**Pull request by antoine-levitt**

### CODE REVIEW

File: src/PlaneWaveBasis.jl [hunk 1]
```diff
@@ -198,20 +197,16 @@ end
     ksymops_thisproc = ksymops[krange_thisproc]
 
     # Setup fft_size and plans
-    (ipFFT_unnormalized,  opFFT_unnormalized,
-     ipBFFT_unnormalized, opBFFT_unnormalized) = build_fft_plans(T, fft_size)
+    (ipFFT,  opFFT, ipIFFT, opIFFT) = build_fft_plans(T, fft_size)
```
Comment (mfherbst) [inline src/PlaneWaveBasis.jl:200]: spacing

File: src/PlaneWaveBasis.jl [hunk 2]
```diff
@@ -65,16 +65,15 @@ struct PlaneWaveBasis{T <: Real}
...
 
     ## Plans for forward and backward FFT
-    # These plans follow DFTK normalization conventions (see above)
+    # All these plans are *completely unnormalized* (eg FFT * IFFT != I)
+    # The normalizations are performed in G_to_r/r_to_G according to
+    # the DFTK conventions (see above)
     opFFT   # out-of-place FFT plan
     ipFFT   # in-place FFT plan
-    opIFFT  # inverse plans
+    opIFFT  # inverse plans (unnormalized; BFFT in FFTW terminology)
```
Comment (mfherbst) [inline src/PlaneWaveBasis.jl:73]: use `opBFFT` and `ipBFFT` ?

File: src/PlaneWaveBasis.jl [hunk 3]
```diff
@@ -438,20 +433,24 @@ In-place version of `G_to_r`.
 @timing_seq function G_to_r!(f_real::AbstractArray3, basis::PlaneWaveBasis,
                              f_fourier::AbstractArray3)
     mul!(f_real, basis.opIFFT, f_fourier)
+    f_real .*= basis.G_to_r_normalization
```
Comment (mfherbst) [inline src/PlaneWaveBasis.jl:436]: Use 5-argument version of `mul!` ?

File: src/PlaneWaveBasis.jl [hunk 4]
```diff
@@ -438,20 +433,24 @@ In-place version of `G_to_r`.
...
     fill!(f_real, 0)
     f_real[kpt.mapping] = f_fourier
 
-    # Perform an FFT
-    mul!(f_real, plan, f_real)
+    # Perform an IFFT
+    mul!(f_real, basis.ipIFFT, f_real)
+    if !skip_normalization
+        f_real .*= basis.G_to_r_normalization
+    end
```
Comment (mfherbst) [inline src/PlaneWaveBasis.jl:452]: 5-argument `mul!` ?

File: src/PlaneWaveBasis.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:200]: what's wrong with the spacing?

File: src/PlaneWaveBasis.jl (hunk 2 above)
Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:73]: yup good point

File: src/PlaneWaveBasis.jl (hunk 3 above)
Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:436]: Not sure FFTW supports it, better to be more explicit here

File: src/PlaneWaveBasis.jl (hunk 4 above)
Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:452]: same as above

File: src/PlaneWaveBasis.jl (hunk 1 above)
Comment (mfherbst) [reply src/PlaneWaveBasis.jl:200]: two spaces after ","

File: src/PlaneWaveBasis.jl (hunk 3 above)
Comment (mfherbst) [reply src/PlaneWaveBasis.jl:436]: I just thought if they will eventually then we don't have to think about it, but I'm pretty sure right now they don't.

File: src/PlaneWaveBasis.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:200]: oof good eye

**Comment by mfherbst** [comment]
I changed the `skip_normalisation=true` interface to `normalisation=false` because it removes the strange double negations.

**Comment by antoine-levitt** [comment]
Normalization feels like a number. Rename to normalize?

**Comment by mfherbst** [comment]
ok

**Pull request merged**